c2rust-analyze/src/main.rs

```diff
@@ -449,13 +449,65 @@ fn run(tcx: TyCtxt) {
             assert_eq!(local, l);
         }
 
+        // Find all calls to `malloc`, `calloc`, `realloc`, and `free`
+        // and track their destination locals as being libc::c_void
+        // pointers to special-case downstream casts
+        for bb_data in mir.basic_blocks().iter() {
+            if let Some(term) = &bb_data.terminator {
+                let term: &Terminator = term;
+                if let TerminatorKind::Call {
+                    ref func,
+                    ref args,
+                    destination,
+                    target: _,
+                    ..
+                } = term.kind
+                {
+                    let func_ty = func.ty(&mir.local_decls, tcx);
+                    use crate::util::Callee::*;
+
+                    let assert_libc_cvoid = |p: &Place<'tcx>| {
```

Comment:
Maybe just name this `assert_is_c_void` as it doesn't have to be `libc`.

Reply:
addressed in 9f72789c